src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -140,19 +140,84 @@ public async Task<SubscriptionState> AssertSubscriptionConnectionDetails(long id
             {
                 var subscription = GetSubscriptionFromServerStore(serverStoreContext, name);
                 var topology = _serverStore.Cluster.ReadDatabaseTopology(serverStoreContext, _db.Name);
+                
                 var whoseTaskIsIt = _db.WhoseTaskIsIt(topology, subscription, subscription);
+                
                 if (whoseTaskIsIt != _serverStore.NodeTag)
                 {
+                    var databaseTopologyAvailabilityExplanation = new Dictionary<string, string>();
+
+                    string generalState = string.Empty;
+                    if (_serverStore.Engine.CurrentState == RachisState.Candidate || _serverStore.Engine.CurrentState == RachisState.Passive)
```

Comment:
Should we extract currentState to local variable? it can change during the execution

Reply:
ok